types/lodash/index.d.ts

```diff
@@ -12269,7 +12269,7 @@ declare namespace _ {
          * @param value The value to inspect.
          * @return Returns true if value is empty, else false.
          */
-        isEmpty(value?: any): boolean;
+        isEmpty(value?: any): value is null | undefined;
```

Comment:
I get your point, and I do don't see this a clean fix either, but didn't find any other alternative, because doing an overload didn't seen right and doing something like `isEmpty(value?: any): (value is null | undefined) | boolean` is not possible as well.